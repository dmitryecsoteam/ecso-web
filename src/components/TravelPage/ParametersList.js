import React from 'react';
import { defaultState } from '../../store/ReduxStore';
import classNames from 'classnames';
import TrackVisibility from 'react-on-screen';


export default ({ destination }) => {

    const parameters = [];
    for (let parameter in defaultState.searchForm.parametersValue) {

        const dots = [];
        for (let i = 1; i <= 5; i++) {
            const className = classNames(
                'parameters-list__dot',
                { 'parameters-list__dot--active': i <= destination[parameter.toLowerCase() + 'Rating'] }
            );
            dots.push(<span key={i} className={className}></span>);
        }

        const ParameterHeader = ({ isVisible }) => {
            const className = classNames('parameters-list__item-header', { 'parameters-list__item-header--animation': isVisible });
            return <div className={className} >
                <span className="parameters-list__label">{parameter}</span>
                <div className="parameters-list__dots-container">{dots}</div>
            </div>
        };

        parameters.push(<div key={parameter} className="parameters-list__item" >

            <TrackVisibility once>
                <ParameterHeader />
            </TrackVisibility>

            <div className="parameters-list__description">{destination[parameter.toLowerCase() + 'Description']}</div>
        </div>);
    }

    return <div className="parameters-list__container">{parameters}</div>
}