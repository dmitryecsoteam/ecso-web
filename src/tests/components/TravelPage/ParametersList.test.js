import React from 'react';
import { shallow } from 'enzyme';
import ParametersList from '../../../components/TravelPage/ParametersList';

import { destinationForParametersList } from '../../fixtures/destinations';

jest.mock('../../../store/ReduxStore.js');

test('should render ParametersList component', () => {
    const wrapper = shallow(<ParametersList destination={destinationForParametersList} />);
    expect(wrapper).toMatchSnapshot();
});