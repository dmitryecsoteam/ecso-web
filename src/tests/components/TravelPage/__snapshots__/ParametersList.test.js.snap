// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ParametersList component 1`] = `
<div
  className="parameters-list__container"
>
  <div
    className="parameters-list__parameter"
    data-description="One of popular Japan beaches 'Otaru dream beach' can be reached within 30 minutes from Sapporo. "
    key="Beach"
  >
    <span
      className="parameters-list__label"
    >
      BEACH
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="2"
      />
      <span
        className="parameters-list__dot"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
  <div
    className="parameters-list__parameter"
    data-description="Many restaraunts having more that 2000 reviews  (according to 'Tripadvisor')"
    key="Food"
  >
    <span
      className="parameters-list__label"
    >
      FOOD
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="2"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
  <div
    className="parameters-list__parameter"
    data-description="No museums from the TOP-100 and no UNESCO World Heritage objects"
    key="Museum"
  >
    <span
      className="parameters-list__label"
    >
      MUSEUM
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot"
        key="2"
      />
      <span
        className="parameters-list__dot"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
  <div
    className="parameters-list__parameter"
    data-description="Many parks"
    key="Nature"
  >
    <span
      className="parameters-list__label"
    >
      NATURE
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="2"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
  <div
    className="parameters-list__parameter"
    data-description="Large city but not known as a shopping destination"
    key="Shopping"
  >
    <span
      className="parameters-list__label"
    >
      SHOPPING
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="2"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
  <div
    className="parameters-list__parameter"
    data-description="Big city with various places for nightlife"
    key="Nightlife"
  >
    <span
      className="parameters-list__label"
    >
      NIGHTLIFE
    </span>
    <div>
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="1"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="2"
      />
      <span
        className="parameters-list__dot parameters-list__dot--active"
        key="3"
      />
      <span
        className="parameters-list__dot"
        key="4"
      />
      <span
        className="parameters-list__dot"
        key="5"
      />
    </div>
  </div>
</div>
`;
